editor/src/components/canvas/canvas-strategies/strategies/reparent-strategy-helpers.ts

```diff
@@ -43,226 +60,119 @@ import {
-    canvasState,
-    interactionSession,
-    missingBoundsHandling,
-  )
-  if (
-    foundReparentStrategy.strategy === reparentStrategy &&
-    foundReparentStrategy.forcingRequired === isForced
-  ) {
-    return isForced ? 0.5 : 3
-  } else if (foundReparentStrategy.strategy !== 'do-not-reparent' && allowAsFallback) {
-    return 2
-  } else {
-    return 0
-  }
-}
-
-function targetIsValid(
-  newParentPath: ElementPath,
-  targetsToFilterOut: ReparentTargetsToFilter | null,
-  missingBoundsHandling: MissingBoundsHandling,
-): boolean {
-  if (targetsToFilterOut == null) {
-    return true
-  } else {
-    const targetToFilter = targetsToFilterOut[missingBoundsHandling].newParent
-    return !EP.pathsEqual(targetToFilter, newParentPath)
+function reparentStrategyForReparentTarget(
+  targetMetadata: ElementInstanceMetadataMap,
+  target: ReparentTarget,
+  convertToAbsolute: boolean,
+): FindReparentStrategyResult {
+  return {
+    ...reparentStrategyForParent(targetMetadata, target.newParent, convertToAbsolute),
+    target: target,
   }
 }
 
-function findReparentStrategy(
+export function findReparentStrategies(
   canvasState: InteractionCanvasState,
-  interactionSession: InteractionSession,
-  missingBoundsHandling: MissingBoundsHandling,
-): FindReparentStrategyResult {
-  const selectedElements = getTargetPathsFromInteractionTarget(canvasState.interactionTarget)
-  if (
-    selectedElements.length === 0 ||
-    interactionSession.activeControl.type !== 'BOUNDING_AREA' ||
-    interactionSession.interactionData.type !== 'DRAG' ||
-    interactionSession.interactionData.drag == null || // TODO delete this drag nullcheck? do we start the reparent on mouse down or mouse move beyond threshold?
-    interactionSession.interactionData.modifiers.alt
-  ) {
-    return { strategy: 'do-not-reparent' }
-  }
-
-  const filteredSelectedElements = getDragTargets(selectedElements)
-
-  const pointOnCanvas = offsetPoint(
-    interactionSession.interactionData.originalDragStart,
-    interactionSession.interactionData.drag,
-  )
-
-  const cmdPressed = interactionSession.interactionData.modifiers.cmd
-
-  const reparentResult = getReparentTargetUnified(
-    existingReparentSubjects(filteredSelectedElements),
-    pointOnCanvas,
-    cmdPressed,
-    canvasState,
-    canvasState.startingMetadata,
-    canvasState.startingAllElementProps,
-    missingBoundsHandling,
-  )
-
-  const newParentPath = reparentResult.newParent
-
-  const parentStayedTheSame = filteredSelectedElements.some(
-    (e) => EP.parentPath(e) === newParentPath,
-  )
+  cmdPressed: boolean,
+  pointOnCanvas: CanvasPoint,
+): Array<FindReparentStrategyResult> {
+  const metadata = canvasState.startingMetadata
+
+  const reparentSubjects =
+    canvasState.interactionTarget.type === 'INSERTION_SUBJECTS'
+      ? newReparentSubjects()
+      : existingReparentSubjects(
+          getDragTargets(getTargetPathsFromInteractionTarget(canvasState.interactionTarget)), // uhh
```

Comment:
`// uhh` ?

Reply:
it was meant to signify that this could is not production quality 🤣

Reply:
I think it's better if reparentSubjects is a parameter of `findReparentStrategies`, because it kinda feels like the reparent and the insertion codes are leaking and intermixing in here